frontend/src/components/Main.js

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useState } from "react";
+import { useSelector, useDispatch, batch } from "react-redux";
+import { useNavigate } from "react-router-dom";
+import { API_URL } from "../utils/utils";
+import user from "reducers/user";
+import Secrets from "./Secrets";
+
+const Main = () => {
+  const accessToken = useSelector((store) => store.user.accessToken);
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const [secretMessage, setSecretMessage] = useState(null);
+
+  useEffect(() => {
+    if (!accessToken) {
+      navigate("/login");
+    } else {
+      // GET
+      const options = {
+        headers: {
+          Authorization: accessToken,
+        },
+      };
+
+      fetch(`${API_URL}/secrets`, options)
+        .then((res) => res.json())
+        .then((data) => {
+          if (data.secret) {
+            setSecretMessage(data.secret);
+          } else {
+            setSecretMessage("Failed to fetch secret message");
+          }
+        })
+        .catch((error) => {
+          console.log("Secrets fetch error", error);
+        });
+    }
+  }, [accessToken, navigate]);
+
+  const logOutButton = () => {
+    // Clear user data from the Redux store and local storage
+    batch(() => {
+      dispatch(user.actions.setUserId(null));
+      dispatch(user.actions.setAccessToken(null));
+      dispatch(user.actions.setUsername(null));
+      dispatch(user.actions.setError(null));
+    });
+    localStorage.removeItem("persist:root");
```

Comment:
We are also curious about localStorage and how you solved it. We didn't have time to fix that tonight so we left it without. Hope we can talk more the coming week about this week too :)